src/request-handler.cpp

```diff
@@ -159,16 +159,11 @@ namespace drachtio {
     /* cancel running timer */
     timer.cancel();
 
-    if(timeout_ms > 0) {
-      /* update timer */
+    if(timeout_ms >= 0) {
+      // from libcurl 7.88.1-10+deb12u4 does not allow call curl_multi_socket_action or curl_multi_perform in curl_multi callback directly
```

Comment:
are there any docs on this on the libcurl site?

Reply:
https://curl.se/libcurl/c/CURLMOPT_TIMERFUNCTION.html

WARNING: do not call libcurl directly from within the callback itself when the timeout_ms value is zero, since it risks triggering an unpleasant recursive behavior that immediately calls another call to the callback with a zero timeout...